src/test/java/teammates/sqlui/webapi/GetNotificationsActionTest.java

```diff
@@ -0,0 +1,310 @@
+
+    @Test
+    protected void testExecute_withValidUserTypeForNonAdmin_shouldReturnData() {
+        List<Notification> testNotifications = new ArrayList<>();
+
+        loginAsInstructor(GOOGLE_ID);
+
+        int expectedNumberOfNotifications = UNREAD_NOTIFICATION_COUNT + READ_NOTIFICATION_COUNT;
+
+        for (int i = 0; i < expectedNumberOfNotifications; i++) {
+            testNotifications.add(getTypicalNotificationWithId());
+        }
+
+        when(mockLogic.getActiveNotificationsByTargetUser(NotificationTargetUser.INSTRUCTOR))
+                .thenReturn(testNotifications);
+
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.INSTRUCTOR.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+
+        GetNotificationsAction action = getAction(requestParams);
+        JsonResult jsonResult = getJsonResult(action);
+
+        NotificationsData output = (NotificationsData) jsonResult.getOutput();
+        List<NotificationData> notifications = output.getNotifications();
+
+        // should fetch correct number of notifications
+        assertEquals(expectedNumberOfNotifications, notifications.size());
+    }
+
+    @Test
+    public void testExecute_withoutUserTypeForAdmin_shouldReturnAllNotifications() {
+        final int expectedNumberOfNotifications = 5;
+        Notification testNotification = getTypicalNotificationWithId();
+
+        loginAsAdmin();
+
+        List<Notification> testNotifications = new ArrayList<>();
+        for (int i = 0; i < expectedNumberOfNotifications; i++) {
+            testNotifications.add(getTypicalNotificationWithId());
+        }
+
+        when(mockLogic.getAllNotifications()).thenReturn(testNotifications);
+
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, null,
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+
+        GetNotificationsAction action = getAction(requestParams);
+        JsonResult jsonResult = getJsonResult(action);
+
+        NotificationsData output = (NotificationsData) jsonResult.getOutput();
+        List<NotificationData> notifications = output.getNotifications();
+
+        assertEquals(expectedNumberOfNotifications, mockLogic.getAllNotifications().size());
+        assertEquals(expectedNumberOfNotifications, notifications.size());
+
+        NotificationData expected = new NotificationData(testNotifications.get(0));
+        NotificationData firstNotification = notifications.get(0);
+        verifyNotificationEquals(expected, firstNotification);
+
+        // notification's shown attribute should not be updated
+        List<Notification> notificationToCheck =
+                mockLogic.getActiveNotificationsByTargetUser(testNotification.getTargetUser());
+        notificationToCheck.forEach(n -> assertFalse(n.isShown()));
```

Comment:
I think you did not set up a return value for when `getActiveNotificationsByTargetUser` is called, so  `notificationToCheck` is `null` by default and the assertion on `line 169` won't be run.